### PR TITLE
fix: allow more meta

### LINKED ISSUE #561: [meta] allow more properties
right now only `<meta name="x">` is included, and `twitter:` prefixed names are dropped

I propose we allow:
- `<meta property="x">` where `property` can be:
  - `og:type`
  - `product:availability`
  - `product:price.amount`
  - `product:price.currency` (or possibly, just allow `product:*`)
- `<meta name="twitter:labelN">` and `<meta name="twitter:dataN">`

example of page with those properties: https://api.adobecommerce.live/visualcomfort/content/us/p/utopia-small-pendant-kw5531
and when previewed: https://main--adobe-edge--visualcomfort.aem.page/us/p/utopia-small-pendant-kw5531

I'm guessing this will also require changes in pipeline.. is there any reason not to include these properties?

### PR DESCRIPTION
**Pull request by maxakuru**
fix #561

### CODE REVIEW

File: src/html2md.js [hunk 1]
```diff
@@ -84,13 +89,54 @@ function assertValidJSON(str) {
...
     throw Error('metadata size limit exceeded');
   }
   return str;
 }
 
+/**
+ * Check if meta name is allowed:
+ *  - non-reserved
+ *  - not starting with 'twitter:'
+ *    - except 'twitter:label' and 'twitter:data'
```
Comment (tripodsan) [inline src/html2md.js:109]: but the logic in html-pipeline is different:

```
    const attr = name.includes(':') && !name.startsWith('twitter:') ? 'property' : 'name';
```

if use `name` if:
- name does not include a `:`
- or if starts with `twitter:`

File: src/html2md.js [hunk 2]
```diff
@@ -84,13 +89,54 @@ function assertValidJSON(str) {
...
+    || name.startsWith('twitter:label')
+    || name.startsWith('twitter:data')
+  );
+}
+
+/**
+ * Check if meta property is allowed:
+ *  - non-reserved
+ *  - og:type
+ *  - product:*
```
Comment (tripodsan) [inline src/html2md.js:129]: the logic in html-pipeline is different:
```
but the logic in html-pipeline is different:

```
    const attr = name.includes(':') && !name.startsWith('twitter:') ? 'property' : 'name';
```

if use `property` if:
- name includes a `:`, but not starts with `twitter:`

**Review by tripodsan — CHANGES_REQUESTED** [review]
I don't quite understand the logic here....

File: src/html2md.js (hunk 1 above)
Comment (maxakuru) [reply src/html2md.js:109]: I think that is consistent with this change.. the code previously disallowed any `twitter:` properties to be provided, this just allows some through (specifically `twitter:label*` and `twitter:data*`. Those are used for Slack link unfurling, for example:

```
<meta name="twitter:label1" content="Price">
<meta name="twitter:data1" content="799">
<meta name="twitter:label2" content="Availability">
<meta name="twitter:data2" content="In stock">
```

results in:
image

File: src/html2md.js (hunk 2 above)
Comment (maxakuru) [reply src/html2md.js:129]: this is also consistent with the change

**Review by tripodsan — APPROVED** [review]

**Pull request merged**